version: "3.9"

services:
  cw_3:
    build:
      context: .
    image: foltonhill/pchernov_app
    ports:
      - 80:80
#    volumes:
#      - ./docker_config.py:/code/default_config.py
    depends_on:
      pg:
        condition: service_healthy
#      migrations:
#        condition: service_complited_successfully
#    command: python app.py
#  migrations:
#    build:
#      context: .
#    image: foltonhill/pchernov_app
#    volumes:
#      - ./docker_config.py:/code/default_config.py
#    depends_on:
#      pg:
#        condition: service_healthy
#    command: flask db upgrade
  pg:
    image: postgres:latest
    environment:
      POSTGRES_USER: flask_app
      POSTGRES_PASSWORD: flask_app_password
      POSTGRES_DB: flask_app
    volumes:
      - ./pg_data:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 5s
      timeout: 5s
      retries: 5

